python/lance/util/versioning.py

```diff
@@ -59,3 +67,140 @@ def get_version_asof(ds: FileSystemDataset, ts: [datetime, pd.Timestamp, str]) -
         if v["timestamp"] <= ts:
             return v["version"]
     raise ValueError(f"{ts} is earlier than the first version of this dataset")
+
+
+def compute_metric(uri: [Path, str],
+                   metric_func: Callable[[FileSystemDataset], pd.DataFrame],
+                   versions: list = None) -> pd.DataFrame:
+    """
+    Compare metrics across versions of a dataset
+    """
+    import lance
+    if versions is None:
+        versions = lance.dataset(uri).versions()
+    results = []
+    for v in versions:
+        if isinstance(v, dict):
+            v = v["version"]
+        vdf = metric_func(lance.dataset(uri, version=v))
+        vdf["version"] = v
+        results.append(vdf)
+    return pd.concat(results)
+
+
+def diff(uri, start_version: int, end_version: int) -> LanceDiff:
```

Comment:
should we change the parameters to `v1` and `v2`?

`start/end` makes me think this is find all versions from v1 to v2. like `git log v1..v2`

Reply:
sure. changed